ui/components/app/assets/token-list/token-list.tsx

```diff
@@ -147,14 +151,26 @@ export default function TokenList({
             currencyRates,
           });
 
-          // Append processed token with balance and fiat amount
-          tokensWithBalance.push({
-            ...token,
-            balance,
-            tokenFiatAmount,
-            chainId,
-            string: String(balance),
-          });
+          if (hideZeroBalanceTokens) {
+            // only hide zero balance tokens if not native gas token
+            if (token.isNative || balance !== '0') {
+              tokensWithBalance.push({
```

Comment:
If this is the same push in both branches, could maybe be simpler with

```ts
if (!hideZeroBalanceTokens || token.isNative || balance !== '0') {
    // push it
}
```

Reply:
https://github.com/MetaMask/metamask-extension/pull/29058/commits/b6d235d135ff4952bcb65ad6d32c0a4e471f43eb